### PR TITLE
Enable plugins to support specifying source code encoding

### PR DESCRIPTION
**Pull request by rliu-parasoft**
Changes are applied to both the Parasoft Coverage Recorder and the Parasoft Coverage Step:

1. Do fill all available source code encoding items.
2. Do check source code encoding for ensuring it's a supported encoding of the Java platform, see at  [java.nio.charset.Charset](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/charset/Charset.html)
3. Empty constructor of the Parasoft Coverage Step required for Stapler.

UI when let source code encoding in the parasoft coverage recorder:
image

UI when let source code encoding in the pipeline support:
image

UI when source code encoding format error:
image

### CODE REVIEW

**Comment by rliu-parasoft** [comment]
After the plug-in is enabled to support specifying source code encoding:
image

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 1]
```diff
@@ -63,10 +69,11 @@ protected Void run() throws Exception {
             RunResultHandler runResultHandler = new RunResultHandler(run);
             var parasoftCoverageRecorder = new ParasoftCoverageRecorder();
             parasoftCoverageRecorder.setPattern(step.getPattern());
+            parasoftCoverageRecorder.setSourceCodeEncoding(step.getSourceCodeEncoding());
             LogHandler logHandler = new LogHandler(taskListener, PARASOFT_COVERAGE_NAME);
             ParasoftCoverageRecorder.CoverageConversionResult coverageResult = parasoftCoverageRecorder.performCoverageReportConversion(
                     run, workspace, logHandler, runResultHandler);
-            CoverageRecorder recorder = setUpCoverageRecorder(coverageResult.getCoberturaPattern());
+            CoverageRecorder recorder = setUpCoverageRecorder(coverageResult.getCoberturaPattern(), coverageResult.getSourceCodeEncoding());
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:76]: We can get the source code encoding directly from the step.

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageRecorder.java [hunk 2]
```diff
@@ -246,16 +257,22 @@ public String defaultPattern() {
     static class CoverageConversionResult {
         private final String coberturaPattern;
         private final Set<String> generatedCoverageBuildDirs;
+        private final String sourceCodeEncoding;
 
-        public CoverageConversionResult(String coberturaPattern, Set<String> generatedCoverageBuildDirs) {
+        public CoverageConversionResult(String coberturaPattern, Set<String> generatedCoverageBuildDirs, String sourceCodeEncoding) {
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageRecorder.java:262]: The `sourceCodeEncoding` is not related to  `CoverageConversionResult`(which is used to store the result of convert Parasoft report to Cobertura report), we do not need it here.

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 3]
```diff
@@ -63,10 +69,11 @@ protected Void run() throws Exception {
             RunResultHandler runResultHandler = new RunResultHandler(run);
             var parasoftCoverageRecorder = new ParasoftCoverageRecorder();
             parasoftCoverageRecorder.setPattern(step.getPattern());
+            parasoftCoverageRecorder.setSourceCodeEncoding(step.getSourceCodeEncoding());
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:72]: This line is also not necessary.

**Comment by awang-parasoft** [comment]
`sourceCodeEncoding` is a required parameter in pipeline script, but it's actually optional. The user have to specify `recordParasoftCoverage pattern: '**/coverage.xml', sourceCodeEncoding: ''` instead of `recordParasoftCoverage pattern: '**/coverage.xml'`(which is what Code Coverage API does).

Error when using `recordParasoftCoverage pattern: '**/coverage.xml'` in pipeline.

```log
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: 7: Missing required parameter: "sourceCodeEncoding" @ line 7, column 17.
                   recordParasoftCoverage pattern: '**/coverage.xml'
                   ^
```

**Pull request merged**